Get-PwshUpdate.ps1

```diff
@@ -29,35 +54,60 @@ https://github.com/Ba4bes/PWSHUpdateCheck
 .LINK
 https://4bes.nl/2019/01/04/powershell-challenge-check-pwsh-version-and-install/
 
-
 #>
+Function Show-Alert {
+    param (
+        [Parameter()]
+        [String]$AlertText
+    )
+    $Alert = New-Object -comobject wscript.shell
+    $Alert.popup($AlertText, 0, "PowershellCore Update", 32 + 4)
+}
+Function PSVersion {
+    $PSVersionTable.PSVersion.ToString()
+}
 
-Function Get-PwshUpdates {
+Function Get-PwshUpdate {
     param (
         [Parameter()]
         [Switch]$Preview
     )
+    # Check the version, this script can't run from the same Powershell version as it would update.
+    $PSVersionCheck = PSVersion
+    If ($PSVersionCheck -like "*-rc*" -or $PSVersionCheck -like "*Preview*") {
+        if ($null -eq $Preview) {
+            Continue
+        }
+        if ($null -ne $Preview) {
+            Throw "The shell is running in PowerShell Core Preview while trying to install Preview. Please run script in PowerShell Core or Windows PowerShell."
+        }
+    }
+    elseif ($PSVersionCheck -notlike "5*" -and $false -eq $Preview) {
+        Throw "The shell is running in PowerShell Core while trying to install Core. Please run script in PowerShell Core Preview or Windows PowerShell."
+    }
 
-    $PwshName = "PowerShell 6-x"
-
+    $PwshName = "PowerShell [0-9]-x"
     #Get the newest PWSHversion from the Powershell Github Metadata.
     $Metadata = Invoke-RestMethod https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/metadata.json
 
     # Change the tags so they are in the same format as the local version
     $PwshRelease = $Metadata.ReleaseTag -replace '^v'
     if ($Preview) {
-        $PwshRelease = ($Metadata.PreviewReleaseTag -replace '^v') -replace '-rc'
+        $PwshRelease = ($Metadata.PreviewReleaseTag -replace '^v') -replace '-preview'
         $PwshName = $PwshName -replace '-x', '-preview'
     }
```

Comment:
```
else {
    $PwshRelease = $Metadata.ReleaseTag -replace '^v'
}
```
and remove line 94 :)

Reply:
Good call, this is cleaner. I've changed it.